# Inertial parameters referred to joint's frame(!), only for arm + hand
# Format of link parameters [m,pgx,pgy,pgz,Ixx,Ixy,Ixz,Iyy,I,yz,Izz]
# Obtained with transformation of URDF file from $(find franka_gazebo)/test/launch/panda-gazebo.urdf
# and $(find franka_description)/robots/common/franka_robot.xacro 
# or from $(find franka_description)/robots/common/inertial.yaml considering a specific Denavit-Hartenberg parametrization

link_1_params: &link1
  - 4.970684            # m
  - 0.0192614005        # pgx
  - 0.23670397208       # pgy
  - -0.521519194596     # pgz
  - 0.769359115528267   # Ixx
  - -0.00768922789181   # Ixy
  - 0.0018818868790595  # Ixz
  - 0.0639089103047552  # Iyy
  - 0.00566574404666157 # Iyz
  - 0.717956481077387   # Izz

link_2_params: &link2
  - 0.646926
  - -0.002031994566
  - 0.00226100637
  - 0.01857971472
  - 0.00850351162402155
  - 0.0102611018210082
  - 0.00398335888393552
  - 0.0265349919016902
  - -0.0007689361029464
  - 0.028124284712195


link_3_params: &link3
  - 3.228604
  - 0.088844724872
  - -0.214708623208
  - -0.126729164208
  - 0.0564949260140708
  - -0.00548764810656226
  - 0.00824833314067574
  - 0.0182492022925201
  - 0.00437725712183958
  - 0.0528783819996061

link_4_params: &link4
  - 3.587895
  - -0.19076837715
  - -0.09850206933
  - 0.374644408005
  - 0.0676772702508599
  - -0.0039053550262761
  - 0.0277158431736259
  - 0.0775861490525396
  - 0.00164448757736927
  - 0.0323994304244513

link_5_params: &link5
  - 1.225946
  - -0.014653732538
  - 0.047121686402
  - 0.05034347249
  - 0.0394275709580355
  - 0.00460024551756311
  - -0.00151524447332703
  - 0.0108695107628286
  - -0.00216405205209813
  - 0.0314603723252604

link_6_params: &link6
  - 1.666555
  - 0.100241616695
  - -0.017527158935
  - 0.023526756935
  - 0.00248046035817079
  - -0.000103758917218685
  - -0.00152411090288331
  - 0.011794560230239
  - -9.35690973146049e-05
  - 0.010567766133107

link_7_params: &link7
  - 0.735522
  - 0.007735484874
  - -0.003127439544
  - 0.045305948634
  - 0.0153200083909496
  - -0.000395108718315752
  - -0.00167248266178378
  - 0.0128990646124284
  - -0.000548359106408232
  - 0.00490965196736095

gripper_params: &hand
  - 0
  - 0
  - -0
  - 0
  - 0
  - -0
  - -0
  - 0
  - -0
  - 0

backstepping_controller:
  link_1: *link1
  link_2: *link2
  link_3: *link3
  link_4: *link4
  link_5: *link5
  link_6: *link6
  link_7: *link7
  end_effector: *hand

computed_torque_controller:
  link_1: *link1
  link_2: *link2
  link_3: *link3
  link_4: *link4
  link_5: *link5
  link_6: *link6
  link_7: *link7
  end_effector: *hand
